File: clients/client-elixir/test/vexilla_client_test.exs
defmodule VexillaClientTest do
  use ExUnit.Case
  doctest VexillaClient

  test "fetches feature flags" do
    config =
      VexillaClient.create_config(
        "https://streamparrot-feature-flags.s3.amazonaws.com",
        "dev"
      )

    flags =
      VexillaClient.get_flags(config, "features", fn base_url, file_name ->
        HTTPoison.start()
        flags_response = HTTPoison.get!("#{base_url}/#{file_name}.json")
      end)

    should_use_feature =
      VexillaClient.should?(flags, "billing", "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a")

    assert should_use_feature == true

    should_use_working_gradual =
      VexillaClient.should?(
        flags,
        "testingWorkingGradual",
        "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a"
      )

    assert should_use_working_gradual == true

    should_use_non_working_gradual =
      VexillaClient.should?(
        flags,
        "testingNonWorkingGradual",
        "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a"
      )

    assert should_use_non_working_gradual == false
  end
end

File: clients/client-dart/test/vexilla_client_test.dart
import 'package:vexilla_client/vexilla_client.dart';
import 'package:test/test.dart';

void main() {
  group('A group of tests', () {
    test('First Test', () async {
      var client = VexillaClient(
          'dev',
          'https://streamparrot-feature-flags.s3.amazonaws.com',
          'b7e91cc5-ec76-4ec3-9c1c-075032a13a1a');

      var flags = await client.fetchFlags('features.json');

      client.setFlags(flags);

      var shouldGradual = client.should('testingWorkingGradual');
      var shouldNotGradual = client.should('testingNonWorkingGradual');

      expect(shouldGradual, isTrue);
      expect(shouldNotGradual, isFalse);
    });
  });
}
